test/commands/provar/automation/metadata/download.nut.ts

```diff
@@ -1,194 +1,332 @@
-  // it('Metadata should be downloaded for the provided connection and return the success message in json format', () => {
-  //   const result = execCmd<SfProvarCommandResult>(
-  //     `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c RegmainSandbox,RegmainOrg,RegressionOrg`
-  //   ).shellOutput;
-  //   expect(result.stdout).to.deep.equal(metadataDownloadConstants.successMessage);
-  // });
+  it('Metadata should not be downloaded and return the error message as invalid value exists in metadataLevel property', async () => {
+    const ProvarDXPropertiesFilePath1 = path.join(process.cwd(), './provardx-properties.json');
+
+    // Read the config.json file
+    const ProvarDXPropertiesFileData = await fileSystem.readFile(ProvarDXPropertiesFilePath1, { encoding: 'utf8' });
+
+    // Parse the JSON data
+    const ProvarDXPropertiesFileParsed = JSON.parse(ProvarDXPropertiesFileData);
+    ProvarDXPropertiesFileParsed['provarHome'] = path.join(process.cwd(), './ProvarHome').replace(/\\/g, '/');
+    ProvarDXPropertiesFileParsed['projectPath'] = path
+      .join(process.cwd(), './ProvarRegression/AutomationRevamp')
+      .replace(/\\/g, '/');
+    setNestedProperty(ProvarDXPropertiesFileParsed, 'metadata.metadataLevel', 'xyz');
+
+    // Convert the updated JSON object back to a string
+    const updatedPropertiesFileData = JSON.stringify(ProvarDXPropertiesFileParsed, null, 4);
+
+    // Write the updated JSON string back to the config.json file
+    await fileSystem.writeFile(ProvarDXPropertiesFilePath1, updatedPropertiesFileData, 'utf8');
 
-  it('Metadata should not be downloaded as incorrect connection name and return the error message', () => {
     const result = execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c RegOrg`
+      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} --connections RegmainOrg`
     ).shellOutput;
     expect(result.stderr).to.include(DOWNLOAD_ERROR);
   });
 
-  it('Metadata should not be downloaded as incorrect connection name and return the error message in json format', () => {
+  it('Metadata should not be downloaded and return the json error message as invalid value exists in metadataLevel property', async () => {
     const result = execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} --connections REGMAINORG --json`
+      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c RegmainOrg --json`
     ).jsonOutput;
     expect(result?.result.success).to.deep.equal(false);
-    // eslint-disable-next-line
+
     expect((result?.result.errors?.[0] as any)?.code).to.equals('DOWNLOAD_ERROR');
   });
 
-  it('Metadata should not be downloaded for the invalid user name and return the error', () => {
+  it('It should Delete the PROVARDX_PROPERTIES_FILE_PATH in sf config.json', async () => {
+    async function deleteFileConfigSfdx(): Promise<void> {
```

Comment:
we have these types of functions as inner functions in multiple NUTs. these seem to be generic functions that can be also used further, so keep these functions outside it().
Make them a function in a separate class instead of creating it as an inner function.

Reply:
This looks like generic function but this need to used only once so added in before